docs/guides/checkpointing_solutions/Emergency_Checkpointing.md

```diff
@@ -0,0 +1,181 @@
+* **Training Slice**: A nodepool with **32 hosts**.
+
+1.  **Estimate Total Checkpoint Size**:
+    `70,000,000,000 parameters × 12 bytes/parameter = 840,000,000,000 bytes`
+    `840,000,000,000 bytes ≈ 840 GB`
+2.  **Calculate Per-Host Checkpoint shard**:
+    `(Total Checkpoint Size / 32 hosts) = 26.25 GB per host`
+
+3.  **Calculate Per-Host Ramdisk Size**:
+    `(Per-Host Checkpoint shard) * 2 = 52.50 GB per host`
+
+4.  **Add a Safety Buffer (e.g., 15%)**:
+    `(Per-Host Ramdisk Size) × 1.15 ≈ 60.3 GB`
+
+In this scenario, you should configure each pod in that slice with a ramdisk of at least **60 GB**.
+
+### Example XPK cluster creation command
+
+```
+OUTPUT_PATH=<gcs bucket output path>
+PROJECT_ID=<project id>
+ZONE=<your zone>
+CLUSTER_NAME=<cluster name>
+TPU_TYPE=<tpu-type>
+MACHINE_TYPE=<cpu machine-type>
+NUM_SLICES=<number of slices>
+RAMDISK_SIZE="60000Mi"
+GKE_VERSION=1.32.3-gke.1785000
+
+gcloud config set project ${PROJECT_ID}
+gcloud config set compute/zone ${ZONE}
+
+git clone https://github.com/AI-Hypercomputer/xpk.git
+
+python3 xpk/xpk.py cluster create \
+--cluster ${CLUSTER_NAME} \
+--cluster-cpu-machine-type=${MACHINE_TYPE} \
+--num-slices=${NUM_SLICES} \
+--tpu-type=${TPU_TYPE} \
+--enable-mtc \
+--enable-gcsfuse-csi-driver \
+--mtc-ramdisk-size=${RAMDISK_SIZE} \
+--mtc-gcs-bucket=${OUTPUT_PATH} \
+--gke-version=${GKE_VERSION}
+
+```
+
+---
+## Maxtext Configuration
+
+This configuration manages a `two-tiered checkpointing` system designed for both durability and rapid recovery.
+
+* **Local Emergency Checkpoints**: It saves checkpoints much more frequently to a fast, local directory on each host (i.e. a ramdisk). If a preemption or failure occurs, the job can restore from this recent local copy almost instantly, minimizing lost work without needing to download from slower persistent storage. This feature is enabled by setting `enable_checkpointing`, `enable_emergency_checkpoint`, `local_checkpoint_directory` and a non-zero `local_checkpoint_period`.
+
+* **Persistent Checkpoints**: These are standard checkpoints saved periodically to durable storage(i.e. GCS bucket). They ensure that you can recover your training state even after a complete job failure. This is controlled by `enable_checkpointing`, and `checkpoint_period`.
+
+
+#### `enable_checkpointing`
+A master switch to enable (`True`) or disable (`False`) saving checkpoints during the training run.
+* **Type**: `boolean`
+* **Default**: `False`
+
+#### `enable_emergency_checkpoint`
+When set to (`True`), this flag enables the two-tiered emergency checkpointing feature. 
+* **Type**: `boolean`
+* **Default**: `False`
+
+#### `async_checkpointing`
+ When set to (`True`), this flag makes checkpoint saving
+    asynchronous. The training step is only blocked for the minimal time needed
+    to capture the model's state, and the actual writing to storage happens in a
+    background thread. This is highly recommended for performance. It's enabled
+    by default.
```

Comment:
This would cause the indented lines to be rendered as a raw-text block:
```suggestion
When set to (`True`), this flag makes checkpoint saving
asynchronous. The training step is only blocked for the minimal time needed
to capture the model's state, and the actual writing to storage happens in a
background thread. This is highly recommended for performance. It's enabled
by default.
```